plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergStatistics.java

```diff
@@ -332,6 +332,57 @@ public void testCollectStatisticsOnWriteToEmptyTable(boolean collectOnStatsOnCre
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test(dataProvider = "testCollectStatisticsOnWriteDataProvider")
+    public void testCollectStatisticsOnWrite(boolean collectOnStatsOnCreateTable, boolean partitioned)
```

Comment:
This is copied from https://github.com/trinodb/trino/pull/17227 and it would be nice to keep as a separate commit

Reply:
Sure, i thought you didn't intend to merge that. Sorry!